test/hwdb-test.sh

```diff
@@ -32,11 +32,9 @@ D=$(mktemp --directory)
 trap "rm -rf '$D'" EXIT INT QUIT PIPE
 mkdir -p "$D/etc/udev"
 ln -s "$ROOTDIR/hwdb" "$D/etc/udev/hwdb.d"
-err=$("$SYSTEMD_HWDB" update --root "$D" 2>&1 >/dev/null)
-if [ -n "$err" ]; then
-    echo "$err"
-    exit 1
-fi
+
+"$SYSTEMD_HWDB" update --root "$D" 2>&1 >/dev/null
+
```

Comment:
The thing is, systemd-hwdb does *not* current exit nonzero on errors. Try it yourself, break `hwdb/60-keyboard.hwdb` in current master, by e. g. adding a space in front of an `evdev` paragraph, and then run `sh -x test/hwdb-test.sh`; you will see that the command succeeds and `$err` is printed. That said, this is surely a bug in systemd-hwdb, so indeed the test should be made more robust. How about something like that:
```
err=$("$SYSTEMD_HWDB" update --root "$D" 2>&1 >/dev/null) || RC=$?
... current if [ -n "$err" ] block as-is ...
[ -z "$RC" ] || exit $RC
```